protocol/flows/relaytransactions/handle_relayed_transactions.go

```diff
@@ -145,42 +145,53 @@ func (flow *handleRelayedTransactionsFlow) broadcastAcceptedTransactions(accepte
 	return flow.Broadcast(inv)
 }
 
-// readMsgTx returns the next msgTx in incomingRoute, and populates invsQueue with any inv messages that meanwhile arrive.
+// readMsgTxOrNotFound returns the next msgTx or msgTransactionNotFound in incomingRoute,
+// returning only one of the message types at a time.
 //
-// Note: this function assumes msgChan can contain only wire.MsgInvTransaction and wire.MsgBlock messages.
-func (flow *handleRelayedTransactionsFlow) readMsgTx() (
-	msgTx *wire.MsgTx, err error) {
+// and populates invsQueue with any inv messages that meanwhile arrive.
+func (flow *handleRelayedTransactionsFlow) readMsgTxOrNotFound() (
```

Comment:
I'm not sure, but I think maybe it should just return wire.Message

Reply:
I prefer two return values over having to do runtime type-checks afterwards